src/lib/hardware/pci.lua

```diff
@@ -134,3 +155,21 @@ function print_device_summary ()
       print(fmt:format(unpack(values)))
    end
 end
+
+function open_usable_devices (options)
+   local drivers = {}
+   for _,device in ipairs(devices) do
+      if #drivers == 0 then
+         if device.usable == 'yes' then
+            print("Opening device "..device.pciaddress)
+            local driver = open_device(device.pciaddress, device.driver)
+            driver:open_for_loopback_test()
+            table.insert(drivers, driver)
+         end
+      end
+   end
+   local options = {devices=drivers,
+                    program=port.Port.loopback_test,
+                    report=true}
+   port.selftest(options)
+end
```

Comment:
This function was intentionally removed in a previous PR #382 . It shouldn't be reintroduced here. AFAIK its not used anywhere.